### PR TITLE
Refs #29548 -- Fixed failing MariaDB 10.3 window tests.

### PR DESCRIPTION
**Pull request by orf**
Ticket: https://code.djangoproject.com/ticket/29548

### CODE REVIEW

**Comment by orf** [comment]
buildbot, test on mariadb.

**Comment by orf** [comment]
buildbot, test on oracle.

**Comment by orf** [comment]
buildbot, test on oracle.

**Comment by orf** [comment]
buildbot, test on oracle.

**Comment by atombrella** [comment]
Thanks for doing this. I remember shortcomings about MariaDB when I implemented the window functions. I'm not sure if adding a note annotation in the documentation for the quirks about `Lead` and `Lag` on MariaDB is out of scope for this PR, or if all those documentation changes should be in one go at a later stage? It's probably much easier to review and quicker to get merged if it's done incrementally.

File: tests/expressions_window/tests.py [hunk 1]
```diff
@@ -416,6 +425,8 @@ def test_lead_offset(self):
         )
 
     def test_lead_default(self):
+        if connection.vendor == 'mysql' and not connection.features.supports_default_in_lead_lag:
```
[the remark this answers is not in the document]
Comment (orf) [reply tests/expressions_window/tests.py]: I didn't know if it was appropriate for this pretty specific case. I've added it now.

File: tests/expressions_window/tests.py [hunk 2]
```diff
@@ -187,6 +187,9 @@ def test_lag(self):
             partition_by=F('department'),
             order_by=[F('salary').asc(), F('name').asc()],
         )).order_by('department')
+        if connection.vendor == 'mysql' and connection.mysql_is_mariadb:
```
[the remark this answers is not in the document]
Comment (orf) [reply tests/expressions_window/tests.py]: Done, I added an `is_mariadb()` method to the test file

**Comment by timgraham** [comment]
If it's feasible.. the thought also crossed my mind.

**Comment by orf** [comment]
It might be for a single window function, but what if there are multiple window functions, perhaps with different ordering? Not sure how current databases handle that.

Perhaps it's better as a separate ticket

**Pull request merged**